climada/util/lines_polys_handler.py

```diff
@@ -0,0 +1,811 @@
+            'orientation': "horizontal"
+            }
+    if 'legend' not in kwargs:
+        kwargs['legend'] = True
+    gdf_plot = gpd.GeoDataFrame(imp_geom.geom_exp)
+    gdf_plot['impact'] = imp_geom.eai_exp
+    if centered:
+        xmin, xmax = u_coord.lon_bounds(imp_geom.coord_exp[:,1])
+        proj_plot = ccrs.PlateCarree(central_longitude=0.5 * (xmin + xmax))
+        gdf_plot = gdf_plot.to_crs(proj_plot)
+    return gdf_plot.plot(column = 'impact', **kwargs)
+
+def exp_geom_to_pnt(exp, res, to_meters, disagg_met, disagg_val):
+    """
+    Disaggregate exposures with (multi-)polygons and/or (multi-)lines
+    geometries to points.
+
+    Parameters
+    ----------
+    exp : Exposures
+        The exposure instance with exp.gdf.geometry containing lines or polygons
+    res : float
+        Resolution of the disaggregation grid / distance. Can also be a
+        tuple of [x_grid, y_grid] numpy arrays. In this case, to_meters is
+        ignored. This is only possible for Polygon-only exposures.
+    to_meters : bool
+       If True, res is interpreted as meters, and geometries are projected to
+       an equal area projection for disaggregation.  The exposures are
+       then projected back to the original projections before  impact
+       calculation. The default is False.
+    disagg_met : string
+        Disaggregation method for the `value` column of the exposure gdf.
+        if 'avg', average value over points
+        if 'fix', same value for each point
+    disagg_val: float
+        if 'None', value is taken from exp.gdf.value column, else the indicated
+        value is taken for disaggregation according to disagg_met
+
+    Returns
+    -------
+    exp_pnt : Exposures
+        Exposures with a double index geodataframe, first level indicating
+        membership of the original geometries of exp,
+        second for the point disaggregation within each geometries.
+
+    """
+
+    line_mask, poly_mask = _line_poly_mask(exp.gdf)
+    gdf_geom = exp.gdf.copy()
+
+    if disagg_val is not None:
+        gdf_geom.value = disagg_val
+
+    if ((disagg_val is None) and ('value' not in gdf_geom.columns)):
+        raise ValueError('There is no value column in the exposure gdf to'+
+                         ' disaggregate from. Please set disagg_val explicitly.')
+
+    if disagg_met not in ['fix', 'avg']:
+        raise ValueError(f"{disagg_met} is not a valid argument for 'disagg_met'."+
+                         " Please choose one of ['fix', 'avg'] ")
+
+    gdf_pnt = gpd.GeoDataFrame([])
+    if np.any(poly_mask):
+        gdf_pnt = gpd.GeoDataFrame(
+            pd.concat(
+                [gdf_pnt, gdf_poly_to_pnt(gdf_geom[poly_mask], res, to_meters, disagg_met)]
+            ))
+    if np.any(line_mask):
+        gdf_pnt = gpd.GeoDataFrame(
+            pd.concat(
+            [gdf_pnt, gdf_line_to_pnt(gdf_geom[line_mask], res, to_meters, disagg_met)]
+            ))
+
```

Comment:
I works now.

Reply:
Actually, this was on purpose. When a dataframe is empty but has an index (e.g., after being masked with an empty mask), then `pd.concat` will always give an `Index` instead of a `MultiIndex`. This is unwanted behaviour here.